letsencrypt-nginx/letsencrypt_nginx/nginxparser.py

```diff
@@ -1,56 +1,241 @@
-"""Very low-level nginx config parser based on pyparsing."""
-import string
+"""Very low-level hand rolled nginx config parser, using recursive descent to
+retain all parsed information, so that the file can be reconstituted."""
 
-from pyparsing import (
-    Literal, White, Word, alphanums, CharsNotIn, Forward, Group,
-    Optional, OneOrMore, Regex, ZeroOrMore)
-from pyparsing import stringEnd
-from pyparsing import restOfLine
+import StringIO
 
+class ParseException(Exception):
+    """A parsing exception happened."""
+    def __init__(self, value):
+        self.value = value
+        super(ParseException, self).__init__()
```

Comment:
Actually, I only put it in since your lint configuration errors if it isn't there. I'm fine with leaving it out though.